src/js/edu-benefits/utils/validations.js

```diff
@@ -284,8 +288,33 @@ function isValidDirectDepositPage(data) {
   return isValidField(isValidRoutingNumber, data.bankAccount.routingNumber);
 }
 
+function isValidBenefitsHistoryPage(data) {
+  return data.activeDutyRepaying.value !== 'Y' ||
+    (!isBlankDateField(data.activeDutyRepayingPeriod.from)
+    && !isBlankDateField(data.activeDutyRepayingPeriod.to)
+    && isValidDateRange(data.activeDutyRepayingPeriod.from, data.activeDutyRepayingPeriod.to));
+}
+
+function isValidRotcScholarshipAmount(data) {
+  return isValidField(isValidMonetaryValue, data.amount) && isValidField(isValidYear, data.year);
+}
+
+function isValidRotcHistoryPage(data) {
+  return data.seniorRotc.rotcScholarshipAmounts.every(isValidRotcScholarshipAmount);
+}
+
 function isValidForm(data) {
-  return isValidBenefitsInformationPage(data);
+  return isValidBenefitsInformationPage(data)
```

Comment:
You mean iterating through the React fields? I think that would be tough to get working.

Or something else?